tensorboard/plugins/interactive_inference/witwidget/notebook/jupyter/js/lib/wit.js

```diff
@@ -0,0 +1,163 @@
+            'tf-interactive-inference-dashboard');
+        this.view_.local = true;
+        this.el.appendChild(this.view_);
+
+        // Add listeners for changes from python.
+        this.model.on('change:examples', this.examples_changed, this);
+        this.model.on('change:config', this.config_changed, this);
+        this.model.on('change:inferences', this.inferences_changed, this);
+        this.model.on('change:eligible_features',
+            this.eligible_features_changed, this);
+        this.model.on('change:mutant_charts', this.mutant_charts_changed, this);
+        this.model.on('change:sprite', this.sprite_changed, this);
+
+        // Add listeners for changes from WIT Polymer element. Passes changes
+        // along to python.
+        this.view_.addEventListener('infer-examples', e => {
+          let i = this.model.get('infer') + 1;
+          this.model.set('infer', i);
+          this.touch();
+        });
+        this.view_.addEventListener('delete-example', e => {
+          this.model.set('delete_example', {'index': e.detail.index});
+          this.touch();
+        });
+        this.view_.addEventListener('duplicate-example', e => {
+          this.model.set('duplicate_example', {'index': e.detail.index});
+          this.touch();
+        });
+        this.view_.addEventListener('update-example', e => {
+          this.model.set('update_example',
+              {'index': e.detail.index, 'example': e.detail.example});
+          this.touch();
+        });
+        this.view_.addEventListener('get-eligible-features', e => {
+          let i = this.model.get('get_eligible_features') + 1;
+          this.model.set('get_eligible_features', i);
+          this.touch();
+        });
+        this.view_.addEventListener('infer-mutants', e => {
+          e.detail['infer_mutants_counter'] = this.inferMutantsCounter++;
+          this.model.set('infer_mutants', e.detail);
+          this.mutantFeature = e.detail.feature_name;
+          this.touch();
+        });
+        setTimeout(()=> {
+          this.config_changed();
+          this.examples_changed();
+          this.sprite_changed();
+        }, 0);
+    },
+
+    // Callback functions for when changes made on python side.
+    examples_changed: function() {
+      const examples = this.model.get('examples');
+      if (!this.view_.updateExampleContents_) {
+        setTimeout(() => this.examples_changed(), 100);
+        return;
+      }
+      if (examples && examples.length > 0) {
+        this.view_.updateExampleContents_(examples, false);
+      }
+    },
+    inferences_changed: function() {
+      const inferences = this.model.get('inferences');
+      this.view_.labelVocab = inferences['label_vocab'];
+      this.view_.inferences = inferences['inferences'];
+    },
+    eligible_features_changed: function() {
+      const features = this.model.get('eligible_features');
+      this.view_.partialDepPlotEligibleFeatures = features
```

Comment:
nit: missing trailing semi-colon (for consistency)

Reply:
done